bundles/framework/layerlist/view/LayerViewTabs/LayerList/LayerCollapse/LayerCollapseHandler.js

```diff
@@ -82,16 +85,82 @@ class ViewHandler extends StateHandler {
         setTimeout(() => this.sandbox.postRequestByName('RemoveMapLayerRequest', [id]), ANIMATION_TIMEOUT);
     }
 
+    // Check if we need to show confirm dialog when activating a whole group
+    showWarn (group) {
+        if (group.layers.length > 9) {
+            if (!group.id || this.state.showWarn.includes(group.id)) {
+                return;
+            }
+            const showWarn = [...this.state.showWarn, group.id];
+            // set groups id to showWarn array so we show the confirm dialog
+            this.updateState({ showWarn });
+        } else {
+            return false;
+        }
+    }
+
+    deactivateGroup (group) {
+        if (this.state.showWarn.includes(group.id)) {
+            const index = this.state.showWarn.indexOf(group.id);
+            if (index === -1) {
+                return;
+            }
+            const showWarn = [...this.state.showWarn];
+            showWarn.splice(index, 1);
+            this.updateState({ showWarn });
+        }
+    }
+
+    // active all layers in selected group
+    addGroup (group) {
+        for (var i in group.layers) {
+            if (!group.layers[i]._id || this.state.selectedLayerIds.includes(group.layers[i]._id)) {
+                continue;
+            }
+            const selectedLayerIds = [...this.state.selectedLayerIds, group.layers[i]._id];
+            this.updateState({ selectedLayerIds });
+            setTimeout(() => this.sandbox.postRequestByName('AddMapLayerRequest', [group.layers[i]._id]), ANIMATION_TIMEOUT);
+        }
+        const index = this.state.showWarn.indexOf(group.id);
+        if (index === -1) {
+            return;
+        }
+        const showWarn = [...this.state.showWarn];
+        showWarn.splice(index, 1);
+        this.updateState({ showWarn });
+    }
+
+    // deactivate all layers in group
+    removeGroup (group) {
```

Comment:
Rename to something like removeGroupLayersFromMap(group) or something descriptive like it.